lib/nexpose/common.rb

```diff
@@ -120,23 +120,89 @@ class Schedule
     attr_accessor :start_month
     attr_accessor :timezone
     attr_accessor :next_run_time
-    attr_accessor :template
 
-    def initialize(type, interval, start, enabled = true)
+    # scan-schedule attributes
+    attr_accessor :repeater_type
+    attr_accessor :scan_template_id
+
+    # @param [Time] start
+    def initialize(type, interval, start, enabled = true, scan_template_id = 'full-audit-without-web-spider')
       @type = type
       @interval = interval
       @start = start
       @enabled = enabled
+      @scan_template_id = scan_template_id
     end
 
     def self.from_hash(hash)
-      schedule = new(hash[:type], hash[:interval], hash[:start])
-      hash.each do |k, v|
-        schedule.instance_variable_set("@#{k}", v)
+      start = nil
+      start = Nexpose::ISO8601.to_time(hash[:start_date]) if hash[:start_date]
+      repeat_scan_hash = hash[:repeat_scan]
+      if repeat_scan_hash.nil?
+        schedule = new('daily', 0, start)
+      else
+        schedule = new(repeat_scan_hash[:type], repeat_scan_hash[:interval], start)
+      end
+      schedule.enabled = hash[:enabled].nil? ? true : hash[:enabled]
+      schedule.scan_template_id = hash[:scan_template_id] || 'full-audit-without-web-spider'
```

Comment:
This will lead to incorrect behaviour as the default template for a schedule should match the template on the site to which it is being added.